scipp-imaging-2D.ipynb

```diff
@@ -8,186 +8,329 @@
+    "    variable = imaging.tiffs_to_variable(to_load)\n",
+    "    variable *= scale_factor\n",
+    "    return variable\n",
     "\n",
-    "from mantid.simpleapi import *\n",
-    "from mantid.api import AlgorithmManager"
+    "ds[\"reference\"] = load_and_scale(folder_name=\"1) R825 Open Beam\", scale_factor=pulse_number_reference)\n",
+    "ds[\"sample\"] = load_and_scale(folder_name=\"2) R825\", scale_factor=pulse_number_sample)\n",
+    "ds[\"sample_elastic\"] = load_and_scale(folder_name=\"3) R825 600 Mpa\", scale_factor=pulse_number_sample_elastic)\n",
+    "ds[\"sample_plastic\"] = load_and_scale(folder_name=\"4) R825 3500 um\", scale_factor=pulse_number_sample_plastic)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    },
+    "scrolled": true
+   },
+   "source": [
+    "## Add TOF and Spectra Coords"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "scrolled": true
+    "pycharm": {
+     "name": "#%%\n"
+    }
    },
    "outputs": [],
    "source": [
-    "# Some helper functions\n",
-    "\n",
-    "def read_x_values(tof_file):\n",
-    "    \"\"\"\n",
-    "    Reads the TOF values from the CSV into a list\n",
-    "    \"\"\"\n",
-    "    tof_values = list()\n",
-    "    with open(tof_file) as fh:\n",
-    "        csv_reader = csv.reader(fh, delimiter='\\t')\n",
-    "        next(csv_reader, None)  # skip header\n",
-    "        for row in csv_reader:\n",
-    "            tof_values.append(float(row[1]))\n",
-    "    return tof_values\n",
-    "\n",
-    "\n",
-    "def _load_tiffs(tiff_dir):\n",
-    "    if not os.path.isdir(tiff_dir):\n",
-    "        raise RuntimeError(tiff_dir + \" is not directory\")\n",
-    "    stack = []\n",
-    "    path_length = len(tiff_dir) + 1\n",
-    "    filenames = sorted(glob.glob(tiff_dir + \"/*.tiff\"))\n",
-    "    nfiles = len(filenames)\n",
-    "    count = 0\n",
-    "    print(f\"Loading {nfiles} files from '{tiff_dir}'\")\n",
-    "    for filename in filenames:\n",
-    "        count += 1\n",
-    "        print('\\r{0}: Image {1}, of {2}'.format(filename[path_length:], count, nfiles), end=\"\")\n",
-    "        img = fabio.open(os.path.join(tiff_dir, filename))\n",
-    "        stack.append(np.flipud(img.data))\n",
-    "\n",
-    "    return np.array(stack)\n",
-    "\n",
-    "def tiffs_to_variable(tiff_dir):\n",
-    "    \"\"\"\n",
-    "    Loads all tiff images from the directory into a scipp Variable.\n",
-    "    \"\"\"\n",
-    "    stack = _load_tiffs(tiff_dir)\n",
-    "    data = stack.astype(np.float64).reshape(stack.shape[0], stack.shape[1]*stack.shape[2])\n",
-    "    return sc.Variable([Dim.Tof, Dim.Spectrum], \n",
-    "                       values=data, variances=data)\n",
+    "# Adds a coordinate for the spectra and TOF\n",
+    "ds.coords[\"spectrum\"] = sc.Variable([\"spectrum\"], values=np.arange(ds[\"sample\"].shape[1]))\n",
```

Comment:
Why is this coord required? It feels like it may have simple been added because "that is what Mantid does"?